tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/object/DistinctObjectTest.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.test.object;
+
+import org.ballerinalang.test.util.BCompileUtil;
+import org.ballerinalang.test.util.CompileResult;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import static org.ballerinalang.test.util.BAssertUtil.validateError;
+
+/**
+ * Test cases for distinct object type definitions.
+ *
+ * @since 2.0
+ */
+public class DistinctObjectTest {
```

Comment:
Shall we add runtime test cases as well?

Reply:
Added in https://github.com/ballerina-platform/ballerina-lang/pull/26098/commits/a7e2849c8dc4a7b749f591b59979b504a6612168